docs/whirlpool/docs/03-Whirlpools SDKs/01-Whirlpools/06-Send Transaction.mdx

```diff
@@ -0,0 +1,372 @@
+      )
+      .await
+      .unwrap();
+      // ...
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+### 4: Create Transaction Message
+To send a transaction on Solana, you need to include a blockhash. A blockhash acts as a timestamp and ensures the transaction has a limited lifetime. Validators use the blockhash to verify the recency of a transaction before including it in a block. A transaction referencing a blockhash is only valid for 150 slots (~1-2 minutes, depending on slot time). After that, the blockhash expires, and the transaction will be rejected.
+
+Here's how the transaction message is created:
+
+<Tabs groupId="programming-languages">
+  <TabItem value="ts" label="Typescript" default>
+    ```tsx title="sendTransaction.ts"
+    const latestBlockHash = await rpc.getLatestBlockhash().send();
+    const transactionMessage = await pipe(
+      createTransactionMessage({ version: 0}),
+      tx => setTransactionMessageFeePayer(wallet.address, tx),
+      tx => setTransactionMessageLifetimeUsingBlockhash(latestBlockHash.value, tx),
+      tx => appendTransactionMessageInstructions(instructions, tx)
+    )
+    ```
+  </TabItem>
+  <TabItem value="rust" label="Rust">
+    ```rust title="main.rs"
+    #[tokio::main]
+    async fn main() {
+      // ...
+      let message = Message::new(
+        &open_position_instructions.instructions,
+        Some(&wallet.pubkey()),
+      );
+      let mut signers: Vec<&dyn Signer> = vec![&wallet];
+      signers.extend(
+        open_position_instructions
+          .additional_signers
+          .iter()
+          .map(|kp| kp as &dyn Signer),
+      );
+      let recent_blockhash = rpc.get_latest_blockhash().await.unwrap();
+      let transaction = Transaction::new(&signers, message, recent_blockhash);
+      // ...
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+### 5: Estimating Compute Unit Limit and Prioritization Fee
+When preparing a transaction, it's important to set a compute unit limit and an appropriate prioritization fee. Setting the compute units too low and the transaction will fail. Setting it too high will make the transaction less favorable for validators to process..
+
+You can get an estimation of the compute units by simulating the transaction on the RPC. To avoid transaction failures caused by underestimating this limit, an additional 100,000 compute units are added, but you can adjust this based on your own tests.
+
+The prioritization fee incentivizes validators to prioritize your transaction, especially during times of network congestion. You can get a list of recently paid prioritization fees, sorting them and selecting a value from that list. In this example, we select the median of the sorted list. The prioritization fee is provided in micro-lamports per compute unit. The estimated total priority fee is calculated as `estimated compute units * prioritization fee`. However, the actual fee charged will depend on the compute units your transaction ultimately consumes.
```

Comment:
Maybe explain here that we take the 50th percentile in this example but that you can play around with it.